rasa_nlu/classifiers/embedding_intent_classifier.py

```diff
@@ -347,8 +354,29 @@ def _tf_sim(self, a, b):
                              "should be 'cosine' or 'inner'"
                              "".format(self.similarity_type))
 
-    def _tf_loss(self, sim, sim_emb):
-        """Define loss"""
+    def _tf_loss(self, sim, sim_emb, similarity_mask):
+        """Define loss
+
+        Arguments:
+        - sim:
+            Tensor of shape [batch_size, num_neg + 1]
+            For each element of the batch, its similarity to the correct
+            intent (first element) and to num_neg wrong intents.
+        - sim_emb:
+            Tensor of shape [batch_size, num_neg]
+            For each element of the batch, the similarity of the correct
+            intent to the num_neg negative intents.
+        - similarity_mask:
+            Tensor of shape [batch_size]
+            Indicates whether the example's similarity to its intent should
+            contribute to the loss.
+        """
+
+        # Set masked similarities to 1
+        sim = tf.concat([
```

Comment:
@georgwiese can you put an assert in here that `sim` is the same before and after this statement if `similarity_mask` is set to all ones (which reflects deactivating the feature right?)

Reply:
I refactored the code so that `similarity_mask` is optional, and `sim` remains unchanged if it is not provided. Also, I now log a warning if a similarity mask is applied. See 44613833dfc7b1687e664035bed754cc773b69e1.